### PR TITLE
Adding cfn test workflow

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Added cfn test workflow
Refactored Project and tested  CFN TEST using cfn-test-helper.sh

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .github/workflows/cfn-test.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          AWS_DEFAULT_REGION: ${{ github.event.inputs.region }}
+          AWS_REGION_SECRET: ${{ secrets.AWS_REGION }}
+          ATLAS_PUBLIC_KEY: ${{ secrets.ATLAS_PUBLIC_KEY }}
+          ATLAS_PRIVATE_KEY: ${{ secrets.ATLAS_PRIVATE_KEY }}
+          ATLAS_ORG_ID: ${{ secrets.ATLAS_ORG_ID }}
+          run: |
+            echo "#############################################################"
+            env
+            cd cfn-resources
+            CFN_FLAGS="--verbose --set-default --region ${AWS_DEFAULT_REGION}" ./cfn-test-helper.sh
```
Comment (andreaangiolillo) [inline .github/workflows/cfn-test.yaml:32]:
```suggestion
            CFN_FLAGS="--verbose --set-default --region ${AWS_DEFAULT_REGION}" ./cfn-testing-helper.sh
```

File: cfn-resources/project/test/cfn-test-create-inputs.sh [hunk 2]
```diff
@@ -21,12 +21,14 @@ mkdir inputs
 #create apikey
 org_id="$ATLAS_ORG_ID"
 
-random_str=$(openssl rand -hex 3)
-api_key_id=$(mongocli iam project apikey create --desc "${CFN_TEST_TAG}" --role GROUP_OWNER --output json | jq -r '.id')
+#random_str=$(openssl rand -hex 3)
```
Comment (andreaangiolillo) [inline cfn-resources/project/test/cfn-test-create-inputs.sh:24]:
```suggestion
```

**Review by andreaangiolillo — COMMENTED** [review]
a left a few comments. No need to wait for my approval once my comments have been addressed. Thanks!

**Review by Zuhairahmed — APPROVED** [review]
LGTM

File: cfn-resources/project/cmd/resource/resource.go [hunk 3]
```diff
@@ -142,13 +138,10 @@ func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
...
-	var name string
-	if currentModel.Name != nil {
-		name = *currentModel.Name
-	}
-	event, model, err := getProject(name, client, currentModel)
+
+	event, model, err := getProjectWithSettings(client, currentModel)
 	if err != nil {
-		return event, err
+		return event, nil
```
[the remark this answers is not in the document]
Comment (govardhanpagidi) [reply cfn-resources/project/cmd/resource/resource.go:144]: event will have the error response already, if we send err response from handler it always return GenericException rather Actual Error Response.

[the remark this answers is not in the document]
Comment (andreaangiolillo) [reply cfn-resources/project/cmd/resource/resource.go:144]: Thanks for the clarification

**Review by ParthasarathyVaradhan — APPROVED** [review]
LGTM

**Pull request merged**